src/System.Management.Automation/engine/hostifaces/History.cs

```diff
@@ -1017,137 +1012,126 @@ public string Id
         /// </summary>
         protected override void EndProcessing()
         {
-            // Invoke-history can execute only one command. If multiple
-            // ids were provided, throw exception
-            if (_multipleIdProvided)
-            {
-                Exception ex =
-                    new ArgumentException
-                    (
-                        StringUtil.Format(HistoryStrings.InvokeHistoryMultipleCommandsError)
-                    );
-
-                ThrowTerminatingError
-                (
-                    new ErrorRecord
-                    (
-                        ex,
-                        "InvokeHistoryMultipleCommandsError",
-                        ErrorCategory.InvalidArgument,
-                        null
-                    )
-                );
-            }
-
             History history = ((LocalRunspace)Context.CurrentRunspace).History;
             Dbg.Assert(history != null, "History should be non null");
 
-            // Get the history entry to invoke
-            HistoryInfo entry = GetHistoryEntryToInvoke(history);
-
-            // Check if there is a loop in invoke-history
-            LocalPipeline pipeline = (LocalPipeline)((LocalRunspace)Context.CurrentRunspace).GetCurrentlyRunningPipeline();
-
-            if (pipeline.PresentInInvokeHistoryEntryList(entry) == false)
+            foreach (var currId in _allIds)
```

Comment:
This will throw with Invoke-History without arguments.

Reply:
Would you like me to add a test for calling `Invoke-History` without any args? At the moment, I don't think there is one.

Reply:
Yes, we need to cover the case by a test.